java/arcs/core/data/TypeUtil.kt

```diff
@@ -0,0 +1,21 @@
+package arcs.core.data
+
+import arcs.core.type.Type
+
+/**
+ * If this Type represents a [SingletonType], [CollectionType], or [EntityType], return the
+ * [Schema] used by the underlying [Entity] that this type represents.
+ */
+fun Type.toSchema(): Schema {
+    when (this) {
+        is SingletonType<*> -> if (this.containedType is EntityType) {
+            return this.containedType.entitySchema
+        }
+        is CollectionType<*> -> if (this.collectionType is EntityType) {
+            return this.collectionType.entitySchema
+        }
+        is EntityType -> return this.entitySchema
+        else -> Unit
+    }
+    throw IllegalArgumentException("Can't get entitySchema of unknown type $this")
+}
```

Comment:
The following might work?
```suggestion
fun Type.toSchema() =  when (this) {
    is SingletonType<*> and containedType is EntityType -> containedType.entitySchema
    is CollectionType<*> and collectionType is EntityType -> collectionType.entitySchema
    is EntityType -> entitySchema
    else ->  throw IllegalArgumentException("Can't get entitySchema of unknown type $this")
}
```

Reply:
Applied with slight modification! 👍